federatedscope/core/communication.py

```diff
@@ -1,19 +1,57 @@
 import grpc
 from concurrent import futures
+import logging
+import math
 
 from federatedscope.core.configs.config import global_cfg
 from federatedscope.core.proto import gRPC_comm_manager_pb2, \
     gRPC_comm_manager_pb2_grpc
 from federatedscope.core.gRPC_server import gRPCComServeFunc
 from federatedscope.core.message import Message
 
-
-class StandaloneCommManager(object):
+logger = logging.getLogger(__name__)
+
+# class StandaloneCommManager(object):
+#     """
+#     The communicator used for standalone mode
+#     """
+#     def __init__(self, comm_queue, monitor=None):
+#         self.comm_queue = comm_queue
+#         self.neighbors = dict()
+#         self.monitor = monitor  # used to track the communication related
+#         # metrics
+
+#     def receive(self):
+#         # we don't need receive() in standalone
+#         pass
+
+#     def add_neighbors(self, neighbor_id, address=None):
+#         self.neighbors[neighbor_id] = address
+
+#     def get_neighbors(self, neighbor_id=None):
+#         address = dict()
+#         if neighbor_id:
+#             if isinstance(neighbor_id, list):
+#                 for each_neighbor in neighbor_id:
+#                     address[each_neighbor] = self.get_neighbors(each_neighbor)
+#                 return address
+#             else:
+#                 return self.neighbors[neighbor_id]
+#         else:
+#             # Get all neighbors
+#             return self.neighbors
+
+#     def send(self, message):
+#         self.comm_queue.append(message)
+#         download_bytes, upload_bytes = message.count_bytes()
+#         self.monitor.track_upload_bytes(upload_bytes)
+
+class StandaloneClientCommManager(object):
```

Comment:
I am not sure whether we have to split the `CommManager` int o`ClientCommManager` and `ServerCommManager` here, maybe we just need `channels` instead of one `channel`